### DISCUSSION TITLE
Can I use PostgreSQL instead of MariaDB?

### DISCUSSION THREAD

**Discussion by iAziz786** [postalserver/postal#1905] [Help with using Postal]
I have self hosted PostgreSQL cluster which I use for almost everything. I was wondering instead of self host MariaDB or MySQL, can I use PosgreSQL? I know this may sound very naive question but if any ORM is getting used that can be possible.

If not, how much effort will be to include the support for it?

**Comment by willpower232** [comment]
At the very least I believe you would need to make your own docker image that had the pg gem in it, I'm not sure how much of the code would need to change so if you're not a rails developer, I'd stick with Maria/MySQL

**Comment by ghevge** [comment]
I am interested in PostgreSQL support too!
 I'm not familiar with rails development, but in Java for example, you have persistence frameworks which allow you to decouple your business logic from the 3rd party DB engines, making it extremely easy to switch in between the DB engines. There are no such persistence frameworks in rails ?
